safe/impact_functions/generic/classified_polygon_people/impact_function.py

```diff
@@ -0,0 +1,411 @@
+                unaffected_geometry = geometry.symDifference(impact_geometry)
+
+                all_affected_geometry.append(impact_geometry)
+
+                # add to the affected area of this area type
+                if area_type not in imp_areas:
+                    imp_areas[area_type] = 0.
+                if area_id not in self.all_affected_areas:
+                    self.all_affected_areas[area_id] = 0.
+                area = impact_geometry.area()
+                imp_areas[area_type] += area
+                self.all_affected_areas[area_id] += area
+
+                self.assign_impact_level(f,
+                                         hazard_id,
+                                         hazard_features,
+                                         unaffected_fields,
+                                         impact_fields,
+                                         unaffected_geometry,
+                                         impact_geometry,
+                                         writer)
+
+                impacted_geometries.append(impact_geometry)
+
+        return impacted_geometries
+
+    def assign_impact_level(self,
+                            f,
+                            hazard_id,
+                            hazard_features,
+                            unaffected_fields,
+                            impact_fields,
+                            unaffected_geometry,
+                            impact_geometry,
+                            writer):
+        """ Assign different impacted areas with their
+        respective level of impact(Affected, Not Affected, Medium)
+
+        :param f: exposure feature
+        :type f: QgsFeature
+
+        :param hazard_id: id of analyzed hazard
+        :type hazard_id: int
+
+        :param hazard_features: List of hazard features
+        :type hazard_features: {}
+
+        :param unaffected_fields: Unaffected fields in the impact layer
+        :type unaffected_fields: {}
+
+        :param impact_fields: Impacted fields in the impact layer
+        :type impact_fields: {}
+
+        :param unaffected_geometry: untouched geometry by the hazard
+         layer
+        :type unaffected_geometry: QgsGeometry
+
+        :param impact_geometry: touched geometry by the hazard layer
+        :type impact_geometry: QgsGeometry
+
+        :param writer: Object of writing the impact layer
+        :type writer: QgsVectorFileWriter
+
+        """
+
+        # Checking the type of provided hazard using
+        # current flood and earthquake distinguishing
+        # attributes
+        try:
+            hazard_attribute = hazard_features[hazard_id].\
+                attribute('h_zone')
+        except KeyError:
+            try:
+                hazard_attribute = hazard_features[hazard_id].\
+                    attribute('FLOODPRONE')
+            except KeyError:
+                hazard_attribute = None
+
+        f_unaffected = QgsFeature(unaffected_fields)
```

Comment:
Use proper variable naming here. See: https://github.com/inasafe/inasafe/wiki/Coding-Standards#naming-conventions.
Check some other variables also in this function.